### ISSUE TITLE
Tracking downloads with plausible and ERDDAP

### ISSUE THREAD

**Issue by JessyBarrette** [HakaiInstitute/hakai-erddap#172]
#166 is getting close to be able to running accordingly and need a few fine tuning and clean. One issue left is to be able to track the data downloads via plausible. 

We can easily deploy scripts via the init.d folder within an erddap container and I think this would be the ideal method to implement the plausible download tracker. 

@fostermh How is setup right now your log scrapper for tracking downloads. Can we include the dependacies within a Dockerfile and add the script via an executable file?

See example of the Dockerfile: [here](https://github.com/HakaiInstitute/hakai-datasets/blob/caprover-deploy/Dockerfile)

and list of executable to run on the container start here: https://github.com/HakaiInstitute/hakai-datasets/tree/caprover-deploy/init.d

Any files within `init.d` set as executable or matching. *.sh will get executed when a container is started.

**Comment by JessyBarrette** [comment]
After talking with @fostermh 

Since the present method @fostermh use is using the appache log on top of the container. We will attempt to log usage by having a second container running in parallel that access the nginx log of the caprover erddap app. 

The initial method was using appache, Matt will review if this method is compatible with NGINX which is used by caprover.

**Comment by JessyBarrette** [comment]
In consequence, we will ignore this present issue while developing #166

**Comment by fostermh** [comment]
expanding on the details Jessy has already posted.

We will attempt to use telegraph running in a docker container to parse the nginx logs from caprover. This mirrors the current production setup in which erddap is running behind a proxy (apache) and telegraph is used to scrape the apache logs and report back on erddap usage.

TODO:

- [ ] export nginx logs from caprover nginx container so other containers can access them. See the following for details:
  - https://github.com/caprover/caprover/issues/880
  - https://caprover.com/docs/nginx-customization.html
  In short, update nginx config, via caprover setup page, to send access logs to` /nginx-shared/`
- [ ] mount nginx logs in telegraph container (`/captain/data/nginx-shared/`) and adjust log scraper settings as needed
- [ ] setup plausible page and add key to telegraph container config
- [ ] setup sentry project and add key to telegraph container config (for cron job like monitoring)
- [ ] adjust sentry cron timeout so we don't get spammed and hate matt
- [ ] profit.

https://github.com/cioos-siooc/cwatch-telegraph

**Comment by JessyBarrette** [comment]
Just to add to present thread. Looks like ERDDAP itself is also suggesting to use the Tomcat/Apache/(NGINX?) log to track statistics and usage. 
https://coastwatch.pfeg.noaa.gov/erddap/download/setup.html#tomcatLogs

@fostermh not sure if you started developing this after reading the ERDDAP docs but this is reassuring to see :)

**Comment by fostermh** [comment]
I had not read the docs but yes nice to see that we independently arrived at the same solution. Hopefully we are on the correct track. :-)

**Comment by fostermh** [comment]
telegraf is scrapping logs for the development version of erddap

- telegraf container has been setup https://captain.erddap.hakai.app/#/apps/details/telegraf
- pushing to plausible https://plausible.server.hakai.app/development.erddap.hakai.app
- and is checking in with sentry https://hakai-institute.sentry.io/crons/hakai-telegraf-erddap/hakai-erddap-telegraf-checkin/

**Comment by fostermh** [comment]
haha spelling corrected. Yes would be good to contrast, sounds good.

**Comment by JessyBarrette** [comment]
I just had a quick look at the plausible and for some reasons the urls it refers to have a section doubled:

```
https://development.erddap.hakai.appdevelopment.erddap.hakai.app/erddap/tabledap/HakaiBamfieldBoL5min.htmlTable
```

Must be something related to the setup with plausible.

**Comment by fostermh** [comment]
analytics are being pushed to google analytics now. closing this issue as we are no longer using plausible. See GA interface for all your reporting fun.

**Comment by fostermh** [comment]
update exploration dashboard which segments ckan vs erddap vs filedownloads from erddap into different tabs
https://analytics.google.com/analytics/web/#/analysis/p463807583/edit/AX_x0b3pSDiSixV006fOug
